### ISSUE TITLE
1248 set filter state

### ISSUE THREAD

**Issue by cicdguy** [insightsengineering/teal#15]
Closes #1248 

Changes the functionality a little bit
- srv in the `FilterStates` has been brought back because `add_filter_state` could no longer insert ui to the shiny because `set_filter_state` is called before UI and servers are initialized (initialization of the objects in FilteredData is done before all $srv module is called in class)
- Due to above I've came back to previous philosophy where new FilterState is added to the reactive field (ReactiveQueue) which triggers reactivity in `FilterStates$srv`
- Insertion of the new card is done only once because of monitoring which `FilterState` is displayed (`private$displayed_states`) and which is in the queue `names(queue$get())`. Difference between queue and displayed is added.

todo: need to update documentation
   
## Working example
Example shows some initial FilterState. I'm keeping this backwards compatible so that for `data.frame` we can still specify filter more directly (`list(SITEID = "CHN-3")`). However I don't like it so much because MAE needs more precise coordinates of the filter (`<FIlterStates name> - <argname/ReactiveQueue> - <varname>`). If we keep this consistent than filter for `data.frame` should rather be `list(list(SITEID = "CHN-3")`.

```r
ADSL:
  SITEID: CHN-3
MAE:
  subjects:
    sex: M
  se1:
    subset:
      StartBP:
      - 22000000
      - 144000000
      WidthBP:
      - 100
      - 800000
    select:
      AGE:
      - 11
      - 18
```

```r
devtools::load_all("~/nest/teal/")
devtools::load_all("~/nest/teal.devel/")
library(random.cdisc.data)
library(hermes)


adsl <- cdisc_dataset("ADSL", radsl(cached = TRUE, na_percentage = 0.2)) %>%
  mutate_dataset(
    "ADSL$SEX[1:20] <- NA
    ADSL$AGE[21:30] <- Inf
    ADSL$AGE[31:40] <- NaN
    ADSL$EOSDT[51:60] <- NA
    ADSL$EOSDT[71:70] <- NA
    ADSL$all_na <- NA
    ADSL$unknown <- as.list(ADSL$SEX)"
  )
adtte <- rcd_cdisc_dataset_connector("ADTTE", radtte, cached = TRUE, ADSL = adsl)  %>%
  mutate_dataset(
    "ADTTE$CNSR <- as.logical(ADTTE$CNSR)
    ADTTE$CNSR[100:110] <- NA"
  )

MAE <- multi_assay_experiment # from hermes
colData(MAE[[1]])$xxx <- 1L
rowData(MAE[[1]])$xxx <- 1L

mae <- dataset("MAE", MAE)
iris <- dataset("IRIS", iris)

data <- cdisc_data(adsl, adtte, iris, mae)

app <- init(
  data = data,
  modules = root_modules(
    static = {
      filter_calls_module()
    }
  ),
  filter = list(
    ADSL = list(SITEID = "CHN-3"),
    MAE = list(
      subjects = list(sex = "M"),
      se1 = list(
        subset = list(
          StartBP = c(22000000L, 144000000L),
          WidthBP = c(100L, 800000L)
        ),
        select = list(
          AGE = c(11L, 18L)
        )
      )
    )
  )
)


shinyApp(app$ui, app$server)
```

Provenance: 
```
Creator: gogonzo
```
